### PR TITLE
Revamp repo and use github.com/shirou/gopsutil

### PR DESCRIPTION
**Pull request by soltzen**
Apologies for the monolithic commit, but it's hard to make atomic commits with such a big change.

Here's why the changes happened: If you run the `cmd/main.go` (which was in `osversion/main.go`before the revamp) on a go 1.18 compiler, you'll get this error:

```
cgo: malformed DWARF TagVariable entry
make[1]: *** [app] Error 2
```

I dug and found that just including `#include <stdlib.h>` is enough to trigger the error with CGO. No idea why this is a problem but it is. It could be because we should do `//go:build darwin && cgo`, not just `//go:build darwin`.

Anyways, I found that less code is always better and shirou's gopsutil library does 95% of what we need, except fetching the Android API / OS Version bit, which is mainly the only addition this library does. At one point, I'll make the changes here upstream to shirou's repo and use everything from there.

Main changes to this repo:

- Add a go.mod
- Remove all unused functions in our codebase
  - We just use `GetHumanReadable()`, not `GetString()` or `GetSemanticVersion`
- Remove all platform-specific functions in lieu of gopsutil
  - Except the Android bits
- Add a working `./cmd` for all platforms
  - The one before it didn't work on Android
- Remove the regex test
  - The test was just checking if `GetString()` was called on a Darwin system, it'll have `darwin` in the string. That job is now delegated to gopsutil, which has an extensive CI test that runs on many platforms. Ours doesn't have a CI test and it's quite redundant to retest the same thing.

### CODE REVIEW

File: osversion_android.go [hunk 1]
```diff
@@ -1,87 +1,47 @@
...
-	defer C.free(unsafe.Pointer(verStr))
-	C.android_get_release(verStr)
-	return C.GoString(verStr), nil
-}
-
-func GetHumanReadable() (string, error) {
-	// First get the semantic versioning scheme
-	verStr, err := GetString()
+func getAndroidAPI() (int, error) {
+	out, err := exec.Command("getprop", "ro.build.version.sdk").Output()
```
Comment (oxtoacart) [inline osversion_android.go:12]: Oh interesting, I didn't know that executing shell commands from an Android program was kosher. Are there any weird gotchas with doing this?

File: cmd/main_android.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	_ "golang.org/x/mobile/app"
+)
+
+func main() {
+	str, err := osversion.GetHumanReadable()
+	if err != nil {
+		log.Printf("Error in osversion.GetHumanReadable: %v", err)
+	}
+	// PINEAPPLE is just to make grepping through logcat easier
+	log.Printf("PINEAPPLE: %s\n", str)
```
Comment (lincolnmantracer) [inline cmd/main_android.go:17]: did you leave this in on purpose?

**Review by lincolnmantracer — APPROVED** [review]
LGTM

File: osversion_android.go (hunk 1 above)
Comment (soltzen) [reply osversion_android.go:12]: Not on an emulator or on my phone (stock OS. Not rooted Google Pixel 5). Both worked just fine.

`getprop` is pretty common, I'd say, so all ROMS would need to support it

File: cmd/main_android.go (hunk 2 above)
Comment (soltzen) [reply cmd/main_android.go:17]: Yep. See the README. This is for testing: https://github.com/getlantern/osversion/blob/9f92664c68c7aecec6efcb3ccb3d0e8856ce5289/README.md#L36

It's just an [egg](https://shellcode.blog/Windows-Exploitation-Egg-hunting/)

**Comment by soltzen** [comment]
I've updated [lantern-desktop](https://github.com/getlantern/lantern-desktop/commit/05f672f3dbed96781a8646bf9badd9c3a6fee614) and [Flashlight](https://github.com/getlantern/flashlight/commit/9eb8abbe036e86b9e72a1a938f29e59f75391676) to follow. I needed to do a minor update to [go-update](https://github.com/getlantern/go-update/commit/06ce639b721d8f4276b62239e47ea0453b4665c4) as well.

**Pull request merged**